packages/ibm-products/src/components/Tearsheet/TearsheetShell.js

```diff
@@ -115,6 +116,23 @@ export const TearsheetShell = React.forwardRef(
     const [depth, setDepth] = useState(0);
     const [position, setPosition] = useState(0);
 
+    // if description exceeds two lines and results ellipsis, then `title` attribute displays the whole text
+    const tooltipTriggerRef = useRef(null);
+    const [tooltipText, setTooltipText] = useState(null);
+
+    useEffect(() => {
+      if (open) {
+        if (
+          tooltipTriggerRef?.current?.scrollHeight >
+          tooltipTriggerRef?.current?.clientHeight
+        ) {
+          setTooltipText(tooltipTriggerRef?.current?.innerHTML);
```

Comment:
i don't think it's necessary to set the text here. you should just be able to use the `description` prop in the tooltip label. i would probably just change this to be a boolean like `useTooltip` instead.

Reply:
This is actually a lot like StringFormatter now. I think we should figure out how we want to handle utilities though.